rest/config_test.go

```diff
@@ -1812,16 +1818,14 @@ func TestSetupDbConfigWithSyncFunction(t *testing.T) {
 					RemoteConfigTlsSkipVerify: true,
 				}
 			}
-			if test.errExpected != nil {
-				test.errExpected = &JavaScriptLoadError{
-					JSLoadType: SyncFunction,
-					Path:       sync,
-					Err:        test.errExpected,
-				}
-			}
 			err := dbConfig.setup(dbConfig.Name, BootstrapConfig{}, nil, nil, false)
 			if test.errExpected != nil {
-				require.True(t, errors.As(err, &test.errExpected)) // nolint: govet // CBG-2242
+				errx509Type := x509.UnknownAuthorityError{}
+				if test.errExpected == errx509Type {
+					require.ErrorContains(t, err, assertX509UnknownAuthority(test.errExpected))
+				} else {
+					require.ErrorContains(t, err, test.errExpected.Error())
+				}
```

Comment:
It still feels odd to have the x509 type checking stuff duplicated in every test.

How about extending `assertX509UnknownAuthority` to cover a non-x509 error but still takes the x509 error type into consideration?

```suggestion
				requireErrorWithX509UnknownAuthority(t, err, test.errExpected.Error())
```

```go
func requireErrorWithX509UnknownAuthority(t testing.TB, actual, expected error) {
	expectedErrorString := expected.Error()

	switch typedError := expected.(type) {
	case x509.UnknownAuthorityError:
		expectedErrorString = typedError.Error()
		if runtime.GOOS == "darwin" {
			expectedErrorString = "certificate is not trusted"
		}
	}

	require.ErrorContains(t, actual, expectedErrorString)
}
```

Reply:
This is actually something similar to what I originally wrote for the last commit but wasn't sure if you wanted me to take the `require.ErrorContains(t, actual, expectedErrorString)` out of the main body of the test into the helper function. I agree this way is much better I'll make the changes now and get the PR updated.